contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,36 @@
+# Managing cStor based OpenEBS Volumes. 
+
+## Background 
+
+
+## Design
+
+Provisioning of cStor based Volume involve two steps. 
+
+1. Creating cStor StoragePools:
+
+- Admin will create SPC that contains information like:
+  - type=block
+  - disks list (type of PDC or specific disks)
+  - nodes list (optional)
+
+- maya-cstor-operator (embedded into maya-apiserver), will be watching for SPCs (type=block). If nodes were not specified in the PVC, the maya-cstor-operator will query the kube-ectd for the availability of the disks and determine the nodes. 
+
+- maya-cstor-operator will create a cStorPoolConfig object -  (with information required to initialize the zpool) 
```

Comment:
will maya-cstor-operator does the disk clean (like formatting/nullifying labels etc) before adding to cstorpoolconfig object (if I understood correctly, this cStorPoolConfig object is a CRD)?

Reply:
Yes. cStorPoolConfig .. changed to cStorPool is the CRD. 

Node maya-cstor-operator doesn't have access to the disks. Can it specify a parameters that can indicate the cStor to format/nullify?

Reply:
And should this information be specified for each disk?